x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentTokenForInitialNodeTests.java

```diff
@@ -0,0 +1,318 @@
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.security.enrollment.EnrollmentTokenGenerator;
+import org.elasticsearch.xpack.security.enrollment.EnrollmentToken;
+import org.elasticsearch.xpack.security.tool.CommandLineHttpClient;
+import org.elasticsearch.xpack.security.tool.HttpResponse;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.elasticsearch.mock.orig.Mockito.doReturn;
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BootstrapPasswordAndEnrollmentTokenForInitialNodeTests extends ESTestCase {
+    private Environment environment;
+
+    @BeforeClass
+    public static void muteInFips(){
+        assumeFalse("Enrollment is not supported in FIPS 140-2 as we are using PKCS#12 keystores", inFipsJvm());
+    }
+
+    @Before
+    public void setupMocks() throws Exception {
+        final Path tempDir = createTempDir();
+        final Path httpCaPath = tempDir.resolve("httpCa.p12");
+        Files.copy(getDataPath("/org/elasticsearch/xpack/security/action/enrollment/httpCa.p12"), httpCaPath);
+        final MockSecureSettings secureSettings = new MockSecureSettings();
+        secureSettings.setString("xpack.http.ssl.truststore.secure_password", "password");
+        secureSettings.setString("xpack.security.http.ssl.keystore.secure_password", "password");
+        secureSettings.setString("keystore.seed", "password");
+        final Settings settings = Settings.builder()
+            .put("xpack.security.enabled", true)
+            .put("xpack.http.ssl.enabled", true)
+            .put("xpack.security.authc.api_key.enabled", true)
+            .put("xpack.http.ssl.truststore.path", "httpCa.p12")
+            .put("xpack.security.http.ssl.enabled", true)
+            .put("xpack.security.http.ssl.keystore.path", "httpCa.p12")
+            .put("xpack.security.enrollment.enabled", "true")
+            .setSecureSettings(secureSettings)
+            .put("path.home", tempDir)
+            .build();
+        environment = new Environment(settings, tempDir);
+    }
+
+    public void testGenerateNewPasswordSuccess() throws Exception {
+        final BootstrapPasswordAndEnrollmentTokenForInitialNode command_util = mock(BootstrapPasswordAndEnrollmentTokenForInitialNode
```

Comment:
we don't usually name our variables like this, we stick to camelCase, so commandUtil